pkg/dataobj/internal/sections/streams/iter.go

```diff
@@ -69,7 +69,7 @@ func IterSection(ctx context.Context, dec encoding.StreamsDecoder, section *file
 		})
 		defer r.Close()
 
-		var rows [1]dataset.Row
+		var rows [3]dataset.Row
```

Comment:
Why was this specifically changed? There may be nice benefits from batching during iteration, but IMO that can be an isolated change in another PR applied to all our iterator types.